src/app/ReadHandler.h

```diff
@@ -170,6 +170,11 @@ class ReadHandler : public Messaging::ExchangeDelegate
         /// @param[in] apReadHandler  ReadHandler getting added
         virtual void OnReadHandlerCreated(ReadHandler * apReadHandler) = 0;
 
+        /// @brief Callback invoked when a ReadHandler's reporting intervals is changed through SetMaxReportingInterval or
+        /// SetMinReportingIntervalForTests
+        /// @param[in] apReadHandler  ReadHandler that change its read intervals
+        virtual void OnReportingIntervalsChanged(ReadHandler * apReadHandler) = 0;
```

Comment:
Apart from tests that are doing something weird, do we need this function?  Normally the reporting intervals cannot change after SubscribeResponse is sent, and until it's sent a handler is not schedule-able, right?  And if we have tests that change the intervals after the tests's equivalent of SubscribeResponse ... we should not have those tests; they are creating codepaths that cannot exist normally.

For now, I would suggest:

1. Remove this new callback.
2. Disable any tests that rely on it, with a follow-up to enable them if it's really needed.  Because again, it sounds like those tests are testing things that should not and cannot happen.